stream-log/src/main/kotlin/io/getstream/log/StreamLog.kt

```diff
@@ -36,45 +36,68 @@ import io.getstream.log.Priority.WARN
 public object StreamLog {
 
     /**
+     * Represent a [StreamLogger] is already installed or not.
      * Let you know if the internal StreamLogger instance used for logs has been initialized or it is using the
-     * default one
+     * default one.
      */
-    public var defaultLoggerOverridden: Boolean = false
+    @JvmStatic
+    public var isInstalled: Boolean = false
         private set
 
     /**
      * [StreamLogger] implementation to be used.
      */
+    @Volatile
     @PublishedApi
     internal var internalLogger: StreamLogger = ErrorStreamLogger
         private set(value) {
-            defaultLoggerOverridden = true
+            isInstalled = true
             field = value
         }
 
     /**
      * [IsLoggableValidator] implementation to be used.
      */
+    @Volatile
     @PublishedApi
     internal var internalValidator: IsLoggableValidator = IsLoggableValidator { priority, _ ->
         priority.level >= ERROR.level
     }
         private set
 
     /**
-     * Sets a [StreamLogger] implementation to be used.
+     * Installs a new [StreamLogger] implementation to be used.
+     */
+    @JvmStatic
+    public fun install(logger: StreamLogger) {
+        synchronized(this) {
+            if (isInstalled) {
+                e("StreamLog") {
+                    "The logger $internalLogger is already installed but you've tried to install a new logger: $logger"
+                }
+            }
+            internalLogger = logger
+        }
+    }
+
+    /**
+     * Uninstall a previous [StreamLogger] implementation.
      */
     @JvmStatic
-    public fun setLogger(logger: StreamLogger) {
-        internalLogger = logger
+    public fun unInstall() {
+        synchronized(this) {
+            internalLogger = SilentStreamLogger
```

Comment:
Shouldn't we update `isInstalled` to `false`??
```suggestion
            internalLogger = SilentStreamLogger
            isInstalled = false
```

Reply:
Good point! We should update `isInstalled` to `false`. 

And what about `SilentStreamLogger`? Do we need to update `internalLogger` to `ErrorStreamLogger` by default?

Reply:
I think on that case is fine to use the `SilentStreamLogger`.
By this method you are "disabling" logs, so you expect no logs

Reply:
Agreed!